pennylane/labs/resource_estimation/ops/identity.py

```diff
@@ -34,6 +34,20 @@ def resource_params(self) -> dict:
     def resource_rep(cls, **kwargs) -> re.CompressedResourceOp:
         return re.CompressedResourceOp(cls, {})
 
+    @staticmethod
+    def adjoint_resource_decomp() -> Dict[re.CompressedResourceOp, int]:
+        return {}
+
+    @staticmethod
+    def controlled_resource_decomp(
+        num_ctrl_wires, num_ctrl_values, num_work_wires
+    ) -> Dict[re.CompressedResourceOp, int]:
+        return {}
+
+    @staticmethod
+    def pow_resource_decomp(z) -> Dict[re.CompressedResourceOp, int]:
+        return {}
+
```

Comment:
I think these should return `{cls.resource_rep(): 1}`. This comment applies to the rest of the codebase as well. I think only the identity's resource decomposition should return `{}` and anything else currently returning `{}` should be changed to return `{re.ResourceIdentity.resource_rep(): 1}`. Users might want to track how many times the identity operator shows up in their circuit, but this can't happen with `return {}`. The performance gain in returning `{}` is probably negligible.

Reply:
Changed 👍🏼